search/search_quality/assessment_tool/main_model.cpp

```diff
@@ -224,19 +224,68 @@ void MainModel::OnShowPositionClicked()
 
 bool MainModel::HasChanges() { return m_contexts.HasChanges(); }
 
+bool MainModel::AlreadyInSamples(FeatureID const & id)
+{
+  CHECK(m_selectedSample != kInvalidIndex, ());
+  CHECK(m_selectedSample < m_contexts.Size(), ());
+
+  bool found = false;
+  ForMatchingEntries(m_contexts[m_selectedSample], id, [&](Edits & edits, size_t index)
```

Comment:
Generally, I dislike these lambdas that returns true/false and the loop that checks the return value of lambda for early exit... This is a linear search over a tiny sequence, so I don't see any profit from early exit, really.